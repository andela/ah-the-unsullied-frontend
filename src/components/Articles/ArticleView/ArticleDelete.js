import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import * as articleActions from '../../../actions/articles/articleActions';
import { getArticles } from '../../../actions/ArticleActions/actions';
import '../../../assets/css/articleDetail.scss';

class ArticleDelete extends Component {
  ownerDelete = () => {
    const auth = this.props.auth;
    if (auth.isAuthenticated) {
      const article = this.props.article.article.article;
      const author = article.author;
      const user = auth.user.username;

      if (user === author.username) {
        return (
          <div className="container">
            <p>
              <button className="danger" onClick={this.deleteAction}>
                Delete
              </button>
            </p>
          </div>
        );
      }
    }
  };

  deleteAction = () => {
    const article = this.props.article.article.article;
    const slug = article.slug;
    const { actions } = this.props;
    actions.deleteArticles(slug);
    getArticles();
    this.props.history.push('/');
  };

  render() {
    return <div>{this.ownerDelete()}</div>;
  }
}

ArticleDelete.propTypes = {
  article: PropTypes.object.isRequired,
  auth: PropTypes.object.isRequired,
  actions: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired
};

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(articleActions, dispatch)
  };
}

function mapStateToProps(state) {
  return {
    article: state.articlereducer,
    auth: state.auth
  };
}

export default connect(
  mapStateToProps,
  { mapDispatchToProps, getArticles }
)(ArticleDelete);
