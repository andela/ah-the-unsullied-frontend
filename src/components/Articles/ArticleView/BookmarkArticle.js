import React, { Component } from 'react';
import { connect } from 'react-redux';
import '../../../assets/css/socialshare.scss';
import {
  bookmarkArticle,
  getBookmarkArticles
} from '../../../actions/BookmarkActions';

getBookmarkArticles();

export class BookmarkArticle extends Component {
  state = {
    bookmarked: '#cfcfd1'
  };
  componentWillMount() {
    this.props.getBookmarkArticles();

    this.setState({ bookmarked: '#cfcfd1' });

    setTimeout(() => {
      if (this.props.allbookmarks.length > 0) {
        const current_article_id = this.props.article.article.article['id'];
        const bookmarked_article_id = this.props.allbookmarks.map(art => {
          return art['article_id'];
        });
        const bookmarked_article = bookmarked_article_id.includes(
          current_article_id
        );

        if (bookmarked_article) {
          this.setState({ bookmarked: '#00acee' });
        }
      }
    }, 1000);
  }
  componentWillReceiveProps(nextProps) {
    if (
      nextProps.bookmark === 'You have successfully bookmarked this article'
    ) {
      this.setState({ bookmarked: '#00acee' });
    }
    if (
      nextProps.bookmark ===
      'You have successfully removed this article from your bookmarks'
    ) {
      this.setState({ bookmarked: '#cfcfd1' });
    }
  }
  handleBookmark = () => {
    const { slug } = this.props.article.article.article;
    this.props.bookmarkArticle(slug);
  };

  render() {
    return (
      <div className="likebuttons" id="bookmark">
        <i
          style={{ height: '100px', color: this.state.bookmarked }}
          className="fas fa-2x fa-bookmark"
          onClick={this.handleBookmark}
        />
      </div>
    );
  }
}

export const mapStateToProps = state => {
  return {
    article: state.articlereducer,
    bookmark: state.bookmarkReducer.bookmarks,
    allbookmarks: state.bookmarkReducer.allbookmarks
  };
};

export default connect(
  mapStateToProps,
  { bookmarkArticle, getBookmarkArticles }
)(BookmarkArticle);
