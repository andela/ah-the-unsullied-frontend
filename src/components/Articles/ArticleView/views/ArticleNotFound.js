import React from ''; 

const notFound = () => {
   return(
       <div>
       <h1>404 PAGE NOT FOUND</h1>
       </div>
   )
}

export default notFound;