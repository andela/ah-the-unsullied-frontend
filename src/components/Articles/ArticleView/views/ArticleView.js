import { Row, Col } from 'react-materialize';
import { NavLink } from 'react-router-dom';
import dateformat from 'dateformat';
import renderHTML from 'react-render-html';
import React from 'react';
import '../../../../assets/css/articleDetail.scss';

const articleDetail = props => {
  const article = props.article.article.article;
  const author = article.author;
  const followStatus = author.following;
  const tags = article.tag_list;
  const date = article.created_at;
  const formatDate = dateformat(date, 'dddd, mmmm dS, yyyy');

  const getTags = () =>
    tags.map(item => {
      return <button className="tags-btn">{item}</button>;
    });

  const getImage = () => {
    if (!author.image) {
      const new_image = require('../../../../assets/images/profile.png');
      return new_image;
    } else {
      return author.image;
    }
  };

  const followBtn = () => {
    if (followStatus === null) {
      return (
        <div>
          <button className="follow">Follow</button>
        </div>
      );
    }
  };

  const ownerUpdate = () => {
    const auth = props.auth;
    if (auth.isAuthenticated) {
      const article = props.article.article.article;
      const author = article.author;
      const user = auth.user.username;

      if (user === author.username) {
        return (
          <div className="update-link">
            <NavLink to="/edit-article">
              <i className="material-icons">edit</i>
            </NavLink>
          </div>
        );
      }
    }
  };

  return (
    <div className="container">
      <div>
        <h3>{article.title}</h3>
        <h5 className="description">{article.description}</h5>
        <Row>
          <Col s={1}>
            <img
              src={getImage()}
              className="responsive-img circle escalate"
              height="50px"
              width="50px"
              alt="Avatar"
            />
          </Col>

          <Col s={1}>
            <a href="/">{author.username}</a>
            {followBtn()}
          </Col>
          <Col s={2}>
            {' '}
            <b>{article.read_time}</b>
          </Col>
          <Col>
            <b className="description">{formatDate}</b>
          </Col>
          { ownerUpdate() }
        </Row>
        <div className="bd-image">
          <p>{renderHTML(article.body)}</p>
        </div>
        {getTags()}

        <hr />
        <div />
      </div>
    </div>
  );
};

export default articleDetail;
