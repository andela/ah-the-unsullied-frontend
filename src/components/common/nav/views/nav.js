import React from 'react';
import { NavLink } from 'react-router-dom';

const NavigationView = () =>{
    return(
      <div>
        <NavLink to='/'>Home</NavLink> 
        <NavLink to='/login'>Login</NavLink>
      </div>
    );
};

export default NavigationView;
