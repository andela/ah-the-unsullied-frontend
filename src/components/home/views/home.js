import React from 'react';

const HomeView = () => {
  return (
    <div>
      <h1>{'Welcome to Author\'s Haven'}</h1>
    </div>
  );
};

export default HomeView;
