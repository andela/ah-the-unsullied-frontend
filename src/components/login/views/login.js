import React from "react";
import { Link } from "react-router-dom";

const LoginView = () => {
  return (
    <form>
      <h1> Login </h1>
      <input type="text" name="email" />
      <br />
      <input type="password" name="password" />
      <br />
      <Link to="/">
        <button type="submit">Login</button>
      </Link>
    </form>
  );
};

export default LoginView;
