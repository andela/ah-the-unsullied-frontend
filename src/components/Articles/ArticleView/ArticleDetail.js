import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Col, Preloader, Row } from 'react-materialize';
import {getArticle}from '../../../actions/articles/articleActions';
import ArticleView from './views/ArticleView';
import ArticleDelete from './ArticleDelete';
import Nav from '../../common/nav';
import SocialShare from '../../../components/SocialSharing/shareArticle';
import Rating from '../../../components/Rating/RatingComponent';
import LikeDislike from '../LikeDislikeArticle';
import BookmarkArticle from './BookmarkArticle';

export class ArticleDetail extends Component {
  state = {
    fetched: false
  };

  componentDidMount = () => {
    const slug = this.props.match.params.slug;
    this.props.getArticle(slug).then(() => {
      this.setState({
        fetched: true
      });
    });
  };

  render() {
    if (!this.state.fetched) {
      return (
        <div className="container">
          <Row>
            <Col s={5}>
              <Preloader size="big" />
            </Col>
          </Row>
        </div>
      );
    } else {
      const slug = this.props.article.article.article.slug;
      const rating = this.props.article.article.article.rating;
      const shareTitle = this.props.article.article.article.title;
      const shareSlug = this.props.match.url;
      return (
        <div>
          <Nav />
          <BookmarkArticle {...this.props} />
          <ArticleView {...this.props} />
          <Rating slug={slug} rating={rating} />
          <ArticleDelete {...this.props} />
          <LikeDislike {...this.props} />
          <SocialShare title={shareTitle} slug={shareSlug} />
        </div>
      );
    }
  }
}



ArticleDetail.propTypes = {
  getArticles: PropTypes.func.isRequired,
  getArticle: PropTypes.func.isRequired,
  match: PropTypes.func.isRequired,
  article: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
    article: state.articlereducer,
    auth: state.auth
})

export default connect(
  mapStateToProps,
 {getArticle}
)(ArticleDetail);
