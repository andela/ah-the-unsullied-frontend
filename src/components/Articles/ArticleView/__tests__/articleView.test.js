import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { articleView } from '../views/ArticleView';
import { mapStateToProps } from '../ArticleDetail';

Enzyme.configure({ adapter: new Adapter() });
describe('single article', () => {
  let props;
  let wrapper;
  let wrapperInstance;
  let state;
  beforeEach(() => {
    props = {
      auth: {
        isAuthenticated: true,
        user: {
          username: 'nesh'
        }
      },

      article: {
        title: 'my article',
        description: 'some stuff',
        slug: 'my-article',
        body: 'gsgsgsg',
        dislikes: 1,
        likes: 2,
        liked: false,
        displiked: false,
        is_reported: false,
        created_at: '01-01-2019',
        author: {
          bio: 'i have abio',
          username: 'nesh',
          image: 'hshhshs'
        }
      },

      getArticle: jest.fn(() => {
        Promise.resolve();
      }), 
      ownerUpdate: jest.fn()
    };

    state = {
      auth: {},
      articlereducer: {
        article: {
          article: {}
        }
      }
    };

    wrapper = shallow(<articleView {...props} />);
    wrapperInstance = wrapper.instance();
  });

  it('should render article view component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
