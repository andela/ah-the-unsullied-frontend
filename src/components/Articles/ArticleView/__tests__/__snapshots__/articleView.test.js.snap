// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`single article should render article view component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <articleView
    article={
      Object {
        "author": Object {
          "bio": "i have abio",
          "image": "hshhshs",
          "username": "nesh",
        },
        "body": "gsgsgsg",
        "created_at": "01-01-2019",
        "description": "some stuff",
        "dislikes": 1,
        "displiked": false,
        "is_reported": false,
        "liked": false,
        "likes": 2,
        "slug": "my-article",
        "title": "my article",
      }
    }
    auth={
      Object {
        "isAuthenticated": true,
        "user": Object {
          "username": "nesh",
        },
      }
    }
    getArticle={[MockFunction]}
    ownerUpdate={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "article": Object {
        "author": Object {
          "bio": "i have abio",
          "image": "hshhshs",
          "username": "nesh",
        },
        "body": "gsgsgsg",
        "created_at": "01-01-2019",
        "description": "some stuff",
        "dislikes": 1,
        "displiked": false,
        "is_reported": false,
        "liked": false,
        "likes": 2,
        "slug": "my-article",
        "title": "my article",
      },
      "auth": Object {
        "isAuthenticated": true,
        "user": Object {
          "username": "nesh",
        },
      },
      "getArticle": [MockFunction],
      "ownerUpdate": [MockFunction],
    },
    "ref": null,
    "rendered": null,
    "type": "articleView",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "article": Object {
          "author": Object {
            "bio": "i have abio",
            "image": "hshhshs",
            "username": "nesh",
          },
          "body": "gsgsgsg",
          "created_at": "01-01-2019",
          "description": "some stuff",
          "dislikes": 1,
          "displiked": false,
          "is_reported": false,
          "liked": false,
          "likes": 2,
          "slug": "my-article",
          "title": "my article",
        },
        "auth": Object {
          "isAuthenticated": true,
          "user": Object {
            "username": "nesh",
          },
        },
        "getArticle": [MockFunction],
        "ownerUpdate": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": "articleView",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
